src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ArrayMethod.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+
+namespace System.Reflection.Emit
+{
+    internal sealed class ArrayMethod : MethodInfo
+    {
+        #region Private Data Members
+        private readonly ModuleBuilder _module;
+        private readonly Type _containingType;
+        private readonly string _name;
+        private readonly CallingConventions _callingConvention;
+        private readonly Type _returnType;
+        private readonly Type[] _parameterTypes;
+        #endregion
+
+        #region Constructor
+        // This is a kind of MethodInfo to represent methods for array type of unbaked type
+        internal ArrayMethod(ModuleBuilder module, Type arrayClass, string methodName,
+            CallingConventions callingConvention, Type? returnType, Type[]? parameterTypes)
+        {
+            _returnType = returnType ?? typeof(void);
+            if (parameterTypes != null)
+            {
+                _parameterTypes = new Type[parameterTypes.Length];
+                for (int i = 0; i < parameterTypes.Length; i++)
+                {
+                    ArgumentNullException.ThrowIfNull(_parameterTypes[i] = parameterTypes[i], nameof(parameterTypes));
```

Comment:
change `_parameterTypes[i] = parameterTypes[i]` to just `_parameterTypes[i]`?

Reply:
Well the assignment also needed, doing 2 things in one statement, it was suggested in previously in similar cases

Reply:
Ah I missed the _